### DISCUSSION TITLE
[PorousFlow] Varying aperture as missing dimension 2D in 2D?

### DISCUSSION THREAD

**Discussion by 1runer** [idaholab/moose#24796] [Q&A Modules: General]
Hi,

Probably a question to @WilkAndy or @cpgr.
I read the explanation regarding lower-dimensional representation of fractures [MultiApp Fracture example](https://mooseframework.inl.gov/modules/porous_flow/nomultiapp_flow_through_fractured_media.html), which sounds quite reasonable to me. I wonder if I can translate the 1D in 2D solution also to 2D in 2D assuming the fracture aperture is the missing out-of-plane dimension. 

For example having a 2D rectangular fracture, the missing dimension in MOOSE would be assumed to be '1'. As long as the aperture is constant, I would expect a porosity of '1' and permeability of 'a²/12' should work out. The fluid mass e.g. outflowing on a boundary should just be corrected by the aperture as missing dimension. 

Having an varying aperture in space, the problem becomes more difficult. Following the webpage, I expect, I have to scale the porosity (like 2D in 3D) by 'a * phi' and permeability as '(a² * a) / 12'? 
Accordingly the diffusion gets also scaled by porosity and diffusion coefficient. 
But how would other PorousFlow objects and kernels, like  Dispersion [PorousFlowDispersiveFlux](https://mooseframework.inl.gov/source/kernels/PorousFlowDispersiveFlux.html) or
mass-flux-dependent objects like [PorousFlowSquarePulsePointSource](https://mooseframework.inl.gov/source/dirackernels/PorousFlowSquarePulsePointSource.html), properly scaled? 
It should make a difference if I inject 1 kg/s in a 1m fracture or 1mm-wide fracture.

Best

**Comment by GiudGiud** [comment]
@WilkAndy @cpgr if we could please get assistance on this, it's work happening at INL. Thank you!

**Reply by GiudGiud** [reply]
@JerryLiu2023

**Reply by 1runer** [reply]
Let's push

**Reply by cpgr** [reply]
Oops - I saw this and then forgot about it!

I suppose the kernels should be ok as is, as the mass balance will use the scaled porosity so they should just work?

I also think that your point source mass should be unchanged as once you have scaled the porosity/permeability in the 'scaled' case Darcy's equation is the same. 

I'm pretty sure I have done something similar with a 2D model representing a thin 3D model, and that you get the same spatial distribution using either a scaled porosity/permeability or a scaled injection rate.

You could do two simple models to convince yourself that they are the same - one with a bigger injection, and one with scaled porosity/permeability.

**Reply by 1runer** [reply]
@cpgr Thanks for your answer.

To scale porosity/perm (and each other component of the entire mass balance equation) sounds reasonable to me.
But for the mass source, even after two weeks of thinking, I don't come to a really good solution.
I'm think you are right, to scale poro/perm by aperture or to divide the Dirac by aperture (to get a bigger injection).

But I'm not sure, if this is the 'expected' solution or if we have to scale poro/perm AND a Dirac (like[PorousFlowSquarePulsePointSource](https://mooseframework.inl.gov/source/dirackernels/PorousFlowSquarePulsePointSource.html)) in the same way? 
Isn't any Dirac just another component of the mass balance at some points, so e.g. it has to be scaled too?

I'm a bit confused, as e.g. the colleagues from the [Golem ](https://github.com/ajacquey/golem/)App scale their [Dirac Point Source](https://github.com/ajacquey/golem/blob/master/src/dirackernels/GolemDiracKernelTH.C) according to the elemental dimension with aperture or borehole diameter. 

Please correct me if I am thinking completely wrong.

**Reply by cpgr** [reply]
Stepping back, are you injecting only in the fracture? If so, say your 'real life' is injection 1 kg/s. Then you want to do that in your dirac kernel too, and only scale the porosity and permeability by aperture as you discussed.

**Reply by 1runer** [reply]
Thanks @cpgr for the answer.
Ok, I hopefully got it now. 

Scaling porosity and permeability is similar to increasing the flow resistance by narrowing down my flow path (fracture) from a 1 m aperture (if the out-of-plane dimension is unscaled) to the 'real' aperture. 
As my injection should still be e.g. 1 kg/s into the scaled fracture, it will cause a very high pressure increase as apertures are typically in millimeter range and so 3 orders in magnitude lower. 

But then it is still not fully clear to me, why a Dirac mass source, e.g. 1D in 3D (a injection well in matrix) or 2D in 3D (fracture flow with matrix) is scaled in other Moose-based apps like [Golem](https://github.com/ajacquey/golem/) in [Dirac Point Source](https://github.com/ajacquey/golem/blob/master/src/dirackernels/GolemDiracKernelTH.C) ? I know that the borehole is implemented differently in PorousFlow.

**Reply by cpgr** [reply]
From a quick poke around the source code of Golem, it looks like the scaling they have is to non-dimensionalise the variables rather than an aperture correction.

**Reply by 1runer** [reply]
If you look in [GolemMaterialBase](https://github.com/ajacquey/golem/blob/master/src/materials/GolemMaterialBase.C) is a scaling_factor introduced as material property for a 1D well as cross sectional area and for a 2D fracture as aperture. 
This is reused in the [Dirac ](https://github.com/ajacquey/golem/blob/master/src/dirackernels/GolemDiracKernelTH.C) as a multiplier.

**Reply by cpgr** [reply]
Perhaps they aren't scaling their porosity in their time kernel (if you don't scale porosity, then you would scale the injection rate)?

**Reply by 1runer** [reply]
The [TimeDerivative](https://github.com/ajacquey/golem/blob/master/src/kernels/GolemKernelTimeH.C) is scaled by the scaling_factor as well. 
Maybe that's the reason why is was confused.

But your explanation sounds logic to me. 


I am so interested in the Dirac because I am trying to use a flow and time dependent aperture law. It looks good for longer time steps, but the pressure increase in the very first time steps is difficult to solve and also needs some minimum aperture.
[Radial_fracture_H.i.txt](https://github.com/idaholab/moose/files/12089719/Radial_fracture_H.i.txt)

**Reply by cpgr** [reply]
Maybe @ajacquey could shed some light on the scaling in Golem?

I'm not surprised it is hard to solve in the first time steps, as the pressure will travel a long way along a thin fracture.
